pkg/registry/pod/etcd/etcd_test.go

```diff
@@ -544,10 +507,10 @@ func TestEtcdCreateBinding(t *testing.T) {
 		},
 	}
 	for k, test := range testCases {
-		storage, bindingStorage, _, fakeClient := newStorage(t)
+		storage, bindingStorage, _, server := newStorage(t)
```

Comment:
You need to opaque the body of the loop into
func() {
}()
to make defer work correctly.